### ISSUE TITLE
MedGAN: Medical Image Translation using GANs (2)

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#120]
### MedGAN architecture
* CasNet generator 구조.
  <image src="https://user-images.githubusercontent.com/67627471/131249020-aa85ada9-2406-4f45-bced-3bcceaf1fa9b.png" width="80%" height="80%" />
##
* image-to-image translation은 high dimensional input tensor를 다른 surface appearance를 가진 output tensor에 매치시키는 것으로 볼 수 있음. + 다양한 input modalities.
  * 그래서 기본 building block을 U-blocks이라는 encoder-decoder 구조로 선택함.
  * (어제 U-Net이라고 잘못 씀.. U-Net에서 착안된 것은 맞음.)
  * U-block은 fully convolutional encoder-decoder network임.
* input(source domain image y) -> stack of 8 convolutional layers(필터 사이즈는 각각 64, 128, 256, 512, 512, 512, 512, 512이고,  kernel size 4 × 4에 stride 2임.) -> batch normalization -> Leaky-ReLU activation functions
* decoding path mirrors the encoding architecture.
  * -> enlarges resolution 효과.
  * upsampling path 필터 사이즈는 각각 512, 1024, 1024, 1024, 1024, 512, 256, 128임.
  * 각각의 layers는 ReLU 활성화 함수를 쓰고 마지막 deconvolutional layer에서만 Tanh 씀.
* U-block에는 skip-connection 있음.
  * encoder와 decoder의 mirrored layers 연결함. (예를 들어서 2번째 인코더 층과 7번째 디코더 층)
  * input과 output 사이의 critical한 low level information을 전달해줌.
  * 이거 없으면 병목 층에서 정보 없어져서 output quality에 심각한 degradation 초래함.
##
* 일반적인 image transformation tasks와 달리 medical images는 image translation 과정에서 detailed structures가 lost되거나 왜곡될 수 있음.
  * 그래서 CasNet 구조를 선택함.
  * ResNet에서 착안.
  * 몇몇 U-blocks를 end-to-end 방식에서 concatenate 시킴. (residual block처럼)
  * U-blocks은 ResNet보다 더 많은 conv 층 가짐. (2~4 -> 16개)
  * input 이미지를 output에 연결하기 위해서 identity mapping을 시키지 않고, intermediate skip connection을 써서 low level information을 전달하고 vanishing gradients를 피함.
##
* discriminator에는 PatchGAN 구조 씀.
  *  -> discriminator’s attention is restricted to small image patches which encourage high frequency correctness and enables detailed outputs by the generator
  * utilization of smaller patches -> promotes sharper results
  * 16 × 16 patch size -> two convolutional layers with 64 and 128 spatial filters -> batch normalization -> Leaky-ReLU -> a convolution layer of output dimension 1 -> a sigmoid activation function -> the required confidence probability map (output!)
##
### training
* MedGAN의 학습 파이프라인
  <image src="https://user-images.githubusercontent.com/67627471/131249938-4015ed97-028e-4dca-97c0-6df231288169.png" width="50%" height="50%" />
* cumulative loss function.
  <image src="https://user-images.githubusercontent.com/67627471/131249992-5b8c13ea-b4a7-4d17-bc28-087c24a0134f.png" width="40%" height="40%" />
* 사용한 datasets 예시.
  <image src="https://user-images.githubusercontent.com/67627471/131250027-4302ca47-b469-4382-82a5-ba3a6f8850a8.png" width="80%" height="80%" />
##
### 결과
* Comparison of the effectiveness of different loss functions used within the MedGAN framework
  <image src="https://user-images.githubusercontent.com/67627471/131250082-ac69396a-cb0c-436f-9a45-2c69ca9ecd69.png" width="80%" height="80%" />
* MedGAN이랑 다른 translation frameworks 결과 비교.
  <image src="https://user-images.githubusercontent.com/67627471/131250100-d3735950-ec83-4c5a-8f6a-d798932caba4.png" />
* translation 결과 비교.
  <image src="https://user-images.githubusercontent.com/67627471/131250209-4cb13aff-4d97-4926-9d7e-caf774e12fb6.png" width="80%" height="80%"/>
